packages/wrangler/src/deployment-bundle/node-compat.ts

```diff
@@ -104,29 +94,17 @@ export function getNodeCompatMode({
 		mode = "v2";
 	} else if (nodejsCompat) {
 		mode = "v1";
-	} else if (node_compat) {
+	} else if (legacy) {
 		mode = "legacy";
 	} else {
 		mode = null;
 	}
 
 	return {
-		legacy: node_compat === true,
+		legacy,
```

Comment:
I would love this function to return a single value for clarity.

It looks to me that `legacy`, `nodejsCompat`, and `nodejsCompatv2` can be collapsed into mode.

`experimentalNodejsCompatV2` might be handled in a different functions.